explainer.md

```diff
@@ -53,20 +52,32 @@ See individual sections for more details on these characteristics.
 
 ## Type definitions
 
-The central part of the Typed Objects specification are *type definition objects*, generally called *type definitions* for short. Type definitions describe the fixed structure of a value.
+The central part of the Typed Objects specification are *type definition objects*, generally called *type definitions* for short. Type definitions describe the fixed structure of an instance in memory, specifying a struct type's fields' types.
 
-### Value type definitions
+A type definition has the internal methods `[[Read]]` and `[[Write]]`, which convert between `Value` and internal representations for the type. `[[Write]]` additionally performs a type-specific type check.
 
-In addition to the `Struct` type [described below](#struct-type-definitions), Typed Objects provide a set of builtin types, exposed as functions:
+For `[[Read]]` the abstract behavior is to take in an internal representation `I`, and return an equivalent value `V`.
+
+For `[[Write]]`, the abstract behavior is to take in a value `V` and perform type-specific steps to convert the value into an internal representation, or throw a `TypeError`.
+
+The behavior of `[[Read]]` and `[[Write]]` for all types provided by this specification is described below.
+
+Host environments can provide exotic objects with different implementations of `[[Read]]` and `[[Write]]`.
+
+### Value Types
+
+The following common value type definitions are provided by this spec:
 
     uint8  int8          any
     uint16 int16         string
     uint32 int32 float32 object
-    uint64 int64 float64
+    uint64 int64 float64 ref
+
+These value types provide implementations of the `[[Read]]` and `[[Write]]` internal methods. Calling the value type as a function executes the steps of the internal `[[Write]]` method.
```

Comment:
Since `[[Write]]` has the abstract signature `jsval &rarr; T` and `[[Read]]` has `T &rarr; jsval`, I think we should say that calling a value type as a function (which needs to have signature `jsval &rarr; jsval`) executes `[[Read]]([[Write]](v))`.